batch/batch/worker/worker.py

```diff
@@ -2915,16 +2915,22 @@ async def update():
 
 
 async def async_main():
-    global port_allocator, network_allocator, worker, docker, image_lock
+    global port_allocator, network_allocator, worker, docker, image_lock, CLOUD_WORKER_API, instance_config
 
     image_lock = aiorwlock.RWLock()
     docker = aiodocker.Docker()
 
+    CLOUD_WORKER_API = await GCPWorkerAPI.from_env() if CLOUD == 'gcp' else AzureWorkerAPI.from_env()
+    instance_config = CLOUD_WORKER_API.instance_config_from_config_dict(INSTANCE_CONFIG)
+    assert instance_config.cores == CORES
+    assert instance_config.cloud == CLOUD
+
+    task_manager = aiotools.BackgroundTaskManager()
```

Comment:
I don't see where the task manager is closed on shutdown.

Reply:
Ah, it used to be owned by the worker so it's shutdown in `Worker.shutdown`, but now that I moved it out I should have the worker borrow it instead and shut it down in the main method.

Reply:
Ok so the reason I had initially moved the task_manager out of the Worker was so that I could give it to the `NetworkAllocator` as well. However, that created a mess because the TaskManager shutdown was hard to separate properly from the Worker shutdown because the order is so important. So I undid my change of moving the TaskManager out of the Worker and instead just created a separate task manager for the NetworkAllocator.

I'm going to follow this up with another PR that sorts out the whole shutdown situation and properly use 1 task manager with context managers, but that got pretty big to also be in this PR.